### ISSUE TITLE
Why is pretrained VGGT used for final Pi3 model?

### ISSUE THREAD

**Issue by Nik-V9** [yyfz/Pi3#6]
Hey @yyfz, awesome work!

I had a question regarding the experimental setup. Based on the paper, it looks all the ablation studies and convergence speed analysis plots are based on training from scratch. 

However, it is reported that the final model is trained with weights initialized from VGGT's encoder (which is frozen) and VGGT's alternating attention module. VGGT itself is trained for 150K GD steps and it looks like Pi3 with two stages is further trained for 200K GD steps. This doesn't seem to be training efficient (a major claim)?

I was wondering if there was a reason to use pretrained VGGT over training from scratch for your final model given that your dataset scale is large & diverse and training length is also long?

I would appreciate your thoughts and response on this!

**Comment by yyfz** [comment]
Hi [Nik-V9](https://github.com/Nik-V9), thanks for your question!

In the ablation study, we have fairly and thoroughly validated the effectiveness of all the components we proposed, as well as our major claims, such as training efficiency. We also verified that our model's convergence speed is significantly faster than VGGT's under the same settings.

Regarding whether two-stage training is necessary, I don't believe it is essential. The primary reason we train the first stage at a lower resolution is to quickly check if our training results are promising.

For our final model, we aim to provide a general-purpose model for the community. In this context, loading some pretrained weights is always better than random initialization (just as VGGT chose to load DINO). Furthermore, we do not have access to all of VGGT's training data (as some of it is internal). Therefore, training from scratch might never reach the performance level of VGGT, making it impossible to provide a general-purpose model for the community.

If we had all of VGGT's training data, I would recommend you try training Pi3 from scratch while also incorporating VGGT's multi-task strategy (I believe adding a tracking head with a corresponding tracking loss would suffice). I am confident that the results from this approach would surpass VGGT.

**Comment by yyfz** [comment]
Thank you for your suggestion! We will try to do that if we have spare computing resources.

**Comment by LongWayGuo** [comment]
> > > > Hey [@yyfz](https://github.com/yyfz), I just noticed from your paper that you disable the global pointmap head for the VGGT baseline. Have you ever tried training the VGGT baseline using the global pointmap head turned on?
> > > 
> > > 
> > > Our experiments show that training the VGGT model with the global pointmap and track head enabled yields significantly better performance than a baseline that only predicts camera poses and the local pointmap. Furthermore, by simply incorporating a global pointmap head into our own model, we can outperform the full VGGT architecture.
> > 
> > 
> > [@yyfz](https://github.com/yyfz) 你好, 我看了下pi3里的global_point_head, 也是变换到第一帧作为参考系来计算point的, 如果这样使用不就破坏了pi3最主要的思路, 置换不变性了吗?
> 
> 我们的backbone还是置换不变的，只是那个global point head指定了一个参考帧，这个head只是作为正则，训练完可以直接删掉，和原始模型就没有区别了。

感谢回复.  另我看上面讨论了添加Tracking_head是作用非常大得一个head,  但我看vggt并没有完全开放这部分代码, pi3也没有这部分, 是有过验证这个结论吗?  
而且tracking_head目前是作为一个辅助任务来训练得,  如果我们推理时可以先用特征提取+匹配获取一批可靠得特征关联, 是否能显著提升global_point得一致性呢, 但这种多模态得输入是否就和当前的tracking_head会有很大差别, 这方面能提供一些参考方法么?

**Comment by Nik-V9** [comment]
This paper is relevant to the question regarding flow/dense correspondence supervision: https://www.zhongxiaocong.com/flow3r

**Comment by LongWayGuo** [comment]
> [@LongWayGuo](https://github.com/LongWayGuo) 其实作用也还行，我记得是只要不要只用它的camera token + depth两个任务训练，结果都还行，在camera token + depth的基础上，加上global pointmap 或者 track 监督，提升的都差不多挺大，三个加一起再提升就没咋提升了。
> 
> 你说到的直接在匹配点做特征相似性在RoMav2里有了，应该会是work的，不过感觉得软一点（比如在浅层）。不是很理解你说的多模态输入指的是什么。

好的, 我看RoMav2是1对1的匹配,  但可能pi3需要有N个互相关联的关系, 这个还得再想想怎么做
多模态输入意思就是想PI3X这样的添加位姿/内参/深度等非图像的输入,  特征关联是否也可以用这种方式提前提取好(假设我们可以做得很准确) 用来辅助整个过程.  因为tracking_head这种方式只是辅助任务, 不是一个输入信息.  相比只在训练时辅助,在推理时也有信息输入是否会更硬一些呢? :)

**Comment by yyfz** [comment]
> > [@LongWayGuo](https://github.com/LongWayGuo) 其实作用也还行，我记得是只要不要只用它的camera token + depth两个任务训练，结果都还行，在camera token + depth的基础上，加上global pointmap 或者 track 监督，提升的都差不多挺大，三个加一起再提升就没咋提升了。
> > 你说到的直接在匹配点做特征相似性在RoMav2里有了，应该会是work的，不过感觉得软一点（比如在浅层）。不是很理解你说的多模态输入指的是什么。
> 
> 好的, 我看RoMav2是1对1的匹配, 但可能pi3需要有N个互相关联的关系, 这个还得再想想怎么做 多模态输入意思就是想PI3X这样的添加位姿/内参/深度等非图像的输入, 特征关联是否也可以用这种方式提前提取好(假设我们可以做得很准确) 用来辅助整个过程. 因为tracking_head这种方式只是辅助任务, 不是一个输入信息. 相比只在训练时辅助,在推理时也有信息输入是否会更硬一些呢? :)

把RoMav2的那个特征相似正则拓展到多帧我试过，应该是能明显增加匹配区域的特征相似性的，不过怎么通过正则学习到更好的表征我觉得还是有研究意义的，您可以探索一下！
